src/orng.cpp

```diff
@@ -295,6 +298,7 @@ void orng::submitpart(name oracle, uint64_t id, uint8_t ver, uint8_t idx, string
     eosio::check(!is_paused(), "paused");
     auto oit = oracles_table.require_find(oracle.value, "unknown oracle");
     check(!oit->suspended, "oracle suspended");
+    check(oit->oracle_index == idx, "oracle can only update its assigned index");
```

Comment:
I think we can leave out the parameter uint8_t idx, and replace this check with `uint8_t idx = oit->oracle_index` and rely on the duplicate part check below